src/processor.rs

```diff
@@ -483,15 +475,24 @@ mod tests {
         //     data: vec![0]
         // };
 
-
-
         // concat data to single buffer
         let mut data_vec = vec![0];
+        data_vec.append(
+            &mut (TryInto::<u32>::try_into(title.len()).unwrap().to_le_bytes())
```

Comment:
@ixmorrow need to match how borsh does this, which means adding 4 bytes before strings as a little endian integer specifying the length of the string. So had to do this before title and before review.